ui/src/app/components/apps-panel.tsx

```diff
@@ -0,0 +1,413 @@
+    [RuntimeStatus.Unknown, () => runtime.status]
+  );
+
+  return (
+    <div data-test-id='runtime-cost' style={{ alignSelf: 'center' }}>
+      {text}
+    </div>
+  );
+};
+
+const RuntimeSettingsButton = (props: { onClickRuntimeConf: Function }) => (
+  <Clickable style={{ padding: '0.5em' }} onClick={props.onClickRuntimeConf}>
+    <FlexColumn style={styles.button}>
+      <FontAwesomeIcon icon={faGear} style={styles.buttonIcon} />
+      <div style={styles.buttonText}>Settings</div>
+    </FlexColumn>
+  </Clickable>
+);
+
+const RuntimeStateButton = (props: { workspace: Workspace }) => {
+  const {
+    workspace: { namespace, googleProject },
+  } = props;
+
+  const [status, setRuntimeStatus] = useRuntimeStatus(namespace, googleProject);
+
+  // transition states, so we don't have to wait for polling
+  const [pausing, setPausing] = useState(false);
+  const [resuming, setResuming] = useState(false);
+
+  // when polling updates the status value, clear our transition states
+  useEffect(() => {
+    if (pausing) {
+      setPausing(false);
+    }
+    if (resuming) {
+      setResuming(false);
+    }
+  }, [status]);
+
+  // transition from RUNNING to STOPPED, or STOPPED to RUNNING
+  const toggleRuntimeStatus = () =>
+    switchCase(
+      status,
+      [
+        RuntimeStatus.Running,
+        () => {
+          setPausing(true);
+          setRuntimeStatus(RuntimeStatusRequest.Stop);
+        },
+      ],
+      [
+        RuntimeStatus.Stopped,
+        () => {
+          setResuming(true);
+          setRuntimeStatus(RuntimeStatusRequest.Start);
+        },
+      ]
+    );
+
+  const [icon, buttonText, enabled] = cond(
+    [
+      pausing || status === RuntimeStatus.Stopping,
+      () => [faSyncAlt, 'Pausing', false],
+    ],
+    [
+      resuming || status === RuntimeStatus.Starting,
+      () => [faSyncAlt, 'Resuming', false],
+    ],
+    [status === RuntimeStatus.Stopped, () => [faPlay, 'Resume', true]],
+    [status === RuntimeStatus.Running, () => [faPause, 'Pause', true]],
+    // choose a (disabled) default to show for other states
+    () => [faPause, 'Pause', false]
+  );
+
+  return (
+    <Clickable
+      disabled={!enabled}
+      style={{ padding: '0.5em' }}
+      onClick={toggleRuntimeStatus}
+    >
+      <FlexColumn style={enabled ? styles.button : styles.disabledButton}>
+        <FontAwesomeIcon {...{ icon }} style={styles.buttonIcon} />
+        <div style={styles.buttonText}>{buttonText}</div>
+      </FlexColumn>
+    </Clickable>
+  );
+};
+
+const RuntimeOpenButton = (props: { workspace: Workspace }) => {
+  const { workspace } = props;
+
+  const [showModal, setShowModal] = useState(false);
+  const [notebookNameList, setNotebookNameList] = useState<string[]>([]);
+
+  useEffect(() => {
+    notebooksApi()
+      .getNoteBookList(workspace.namespace, workspace.id)
+      .then((nbl) =>
+        setNotebookNameList(nbl.map((fd) => dropNotebookFileSuffix(fd.name)))
```

Comment:
i wonder if we can move this to a util class not this PR but a clean up PR maybe, since after https://github.com/all-of-us/workbench/pull/7160 the same logic will be used in 3 places